### ISSUE TITLE
[Idea] Create generic winepak flatpak instead of separate flatpak for each app

### ISSUE THREAD

**Issue by Maryse47** [winepak/winepak#17]
I think creating generic flatpak for wine + winetricks or PlayOnLinux/Q4wine would be better approach than creating flatpak for each existing windows app separately.

Users could install specific apps of their choice inside that flatpak similar as they're doing it with wine today but they will be sandboxed from the rest of their system.

They won't be sandboxed from each other but I think wine install prefixes provide some isolation - better than what you have native on windows anyway.

Current approach where creating flatpak for each app is a lot of work and doesn't scale.

**Comment by fabiscafe** [comment]
How would you include this into the stores (gnome software/ kde discover)? How would you deal with special versions of wine you need for some apps? I dont think thats a good idea to start with

**Comment by Maryse47** [comment]
> How would you include this into the stores (gnome software/ kde discover)?

Pretty normal as flatpak for wine/q4wine/PlayOnLinux app. Can be added to flathub even.

> How would you deal with special versions of wine you need for some apps?

Flatpak would include latest stable or latest development wine version. I don't know if flatpak can handle downloading arbitrarily wine versions through q4wine or POL. If app needs some specific patches (?) for wine then it can be packaged separately as currently.

> I dont think thats a good idea to start with

Let's face it - currently there are 12 winepak apps out of thousands supported by wine. That makes the whole project rather meaningless for users. Making 10 times of that would need tremendous amount of work and it will be still a single drop in the sea. My idea is due to pragmatism.

**Comment by fabiscafe** [comment]
Ahh okay so flatpaking Just pol/winehelpers. But then I can't see any benefit over using distros winehelper/wine.

Thing is winepak is a 1man project as I can see. And it would be nice to have flatpaks for wine stuff that's hard to install right now. For a generic wine I don't see any benefit (remember, that uses don't care about sandboxing) over the distro wine version.

It would be nice to have the people of https://lutris.net/ on board, otherwise I don't see a real future for this project, while it is something that is badly needed.

But as always, better leave it than to do it wrong

**Comment by Maryse47** [comment]
> But then I can't see any benefit over using distros winehelper/wine.

Installing wine system-wide requires installing many 32-bit libraries, in Debian you have to add whole i386 repos to apt. Moreover you are left with the same outdated wine version for a few years in LTS distros.

> Thing is winepak is a 1man project as I can see

So it's easier to create one generic flatpak instead of many.

> remember, that uses don't care about sandboxing

I know many people who care about sandboxing which is one of the greatest flatpak features especially for crap proprietary software which is installed through wine so I don't get your point.

People who don't care about sandboxing should use AppImage rather than flatpak which is specifically designed around sandboxing.

**Comment by ghost** [comment]
@Maryse47  But, while winepak is using flatpak, if your bottles/prefixes and game files are outside flatpak do everything still gets sandboxed without issues? Like `--filesystem=/any/base/folder/with/my/games`

I think it would be better to drop PlayOnLinux and, if this is even possible, make it work like:

- flatpak run --command=winepak-cli org.winepak.Winepak --branch=stable --version=3.0.3  --winearch=win64 --exec="wine TombRaider.exe"

- flatpak run --command=winepak-cli org.winepak.Winepak --branch=stable --version=3.0.3  --winearch=win64 --exec="winetricks fontsmooth=rgb gdiplus vcrun2015"

Really bad examples but you get the idea. So winepak could be used as a service that provides "any" wine version you want, with sandboxing. I'm just not sure how composable flatkpak is as I haven't made anything complex with it.

**Comment by Maryse47** [comment]
There is no possibility to run flatpak without sandbox so everything running through flatpak is sandboxed. It would be safer to install wine apps (bottles/prefixes) to `~/.var/app/winepak-cli org.winepak.Winepak` though.

**Comment by paulcarroty** [comment]
> There is no possibility to run flatpak without sandbox so everything running through flatpak is sandboxed.

WHAT?

**Comment by Maryse47** [comment]
> WHAT?

Yep. You can punch holes in the sandbox to access files on host if you want but you can't for example run host binaries. Also this isn't something that may happen by accident - you have to pass it explicitly in flatpak config.

**Comment by paulcarroty** [comment]
https://unix.stackexchange.com/questions/412869/how-to-allow-gui-application-in-flatpak-to-run-cli-tool?rq=1

Just use google.

**Comment by Maryse47** [comment]
I'm maintainer and contributor of several apps in flathub, I don't have to use google to know anything about flatpak. There is new tool called `flatpak-spawn` which can run host commands and is much more appropriate than what you linked but (ab)using this isn't much welcomed in flathub. Also it still has to be specifically allowed in flatpak sandbox.

Anyway the question was: "do everything still gets sandboxed without issues?" and above is totally unrelated. If you have anything in-topic to add please do.

**Comment by paulcarroty** [comment]
@Maryse47 OK, see the issue above, I think to stole your idea. PlayOnLinux/Q4wine seems dead, Lutris in active development.

**Comment by Maryse47** [comment]
PlayOnLinux is in transition ([now Pheonics](https://github.com/PhoenicisOrg/phoenicis)) and they're now proposing the very same idea: https://github.com/winepak/applications/issues/146
Q4Wine:`Latest commit 22adeb7 23 days ago` `v1.3.10  released this on Sep 12`. That's far from dead.
